cross-account-region-data-pipeline/data_processing_pipeline/storage.py

```diff
@@ -0,0 +1,142 @@
+from aws_cdk.aws_events import Rule
+
+class DataStorage(BaseStack):
+
+    def __init__(self, scope: Construct, id: str, environment_id: str, params: Dict, mode: str, compute_params: Dict, **kwargs: Any) -> None:
+        super().__init__(scope, id, environment_id, **kwargs)
+        self._params = params
+        self._environment_id = environment_id
+        self._S3_NAME = params.get('s3BucketName')
+        self._mode = mode
+        self._compute_account = compute_params.get("account")
+        self._compute_region = compute_params.get("region")
+
+        self._raw_bucket = self._create_bucket(name = self._S3_NAME)
+        
+        if self._mode != "same_account_region":
+            self._create_eb_rule()
+
+
+    def _create_bucket(self, name: str) -> Bucket:
+        bucket_key: Key = KMSFactory.key(
+            self,
+            id=f"{name}-bucket-key",
+            environment_id=self._environment_id,
+            description=f"{name.title()} Bucket Key",
+            alias=f"{name}-bucket-key",
+            enable_key_rotation=True,
+            pending_window=cdk.Duration.days(30),
+            removal_policy=cdk.RemovalPolicy.DESTROY,
+        )
+        bucket_key.add_to_resource_policy(
+            statement=PolicyStatement(
+                effect=Effect.ALLOW,
+                principals=[AccountPrincipal(self._compute_account)],
+                actions=[
+                    "kms:CreateGrant",
+                    "kms:Decrypt",
+                    "kms:DescribeKey",
+                    "kms:Encrypt",
+                    "kms:GenerateDataKey*",
+                    "kms:ReEncrypt*"
+                ],
+                resources=["*"],
+            )
+        )
+
+        bucket: Bucket = S3Factory.bucket(
+            self,
+            id=f"{name}-bucket",
+            environment_id = self._environment_id,
+            bucket_name=name,
+            encryption=BucketEncryption.KMS,
+            encryption_key=bucket_key,
+            access_control=BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
+            removal_policy=cdk.RemovalPolicy.RETAIN,
+            event_bridge_enabled= True 
+        )
+        bucket.add_to_resource_policy(
+            permission=PolicyStatement(
+                effect=Effect.ALLOW,
+                principals=[AccountPrincipal(self._compute_account)],
+                actions=[
+                    "s3:Get*"
+                ],
+                resources=[f"{bucket.bucket_arn}/*"],
+            )
+        )
+
+        StringParameter(
+            self,
+            f"{name}-bucket-key-arn-ssm",
+            parameter_name=f"/DDK/KMS/{name.title()}BucketKeyArn",
+            string_value=bucket_key.key_arn,
+        )
+
+        StringParameter(
+            self,
+            f"{name}-bucket-arn-ssm",
+            parameter_name=f"/DDK/S3/{name.title()}BucketArn",
+            string_value=bucket.bucket_arn,
+        )
+
+        return bucket
+
+    def _create_eb_rule(self):
```

Comment:
Nice! I'd say this is the "meat" of this example that enables cross-account cross-region event bus